Tests/BSONTests/CommonTestUtils.swift

```diff
@@ -0,0 +1,107 @@
+@testable import BSON
+import Foundation
+import Nimble
+import XCTest
+
+/// Cleans and normalizes given JSON Data for comparison purposes
+public func clean(json: Data?) throws -> JSON {
+    let jsonDecoder = JSONDecoder()
+    guard let jsonData = json else {
+        fatalError("json should be not nil")
+    }
+    do {
+        let jsonEnum = try jsonDecoder.decode(JSON.self, from: jsonData)
+        return jsonEnum
+    } catch {
+        fatalError("json should be decodable to jsonEnum")
+    }
+}
+
+/// Adds a custom "cleanEqual" predicate that compares Data representing JSON with a JSON string for equality
+/// after normalizing them with the "clean" function
+public func cleanEqual(_ expectedValue: String?) -> Predicate<Data> {
```

Comment:
I think we can just make `expectedValue` non-nil to make the body of this a little cleaner and remove the need for a `fatalError` above.

Reply:
okay, changed this

Reply:
okay